cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -81,13 +93,55 @@ class UploadResultToSw360Command : CliktCommand(
         val sw360Connection = Sw360Storage.createConnection(sw360Config, sw360JsonMapper)
         val sw360ReleaseClient = sw360Connection.releaseAdapter
         val sw360ProjectClient = sw360Connection.projectAdapter
+        val downloader = Downloader(globalOptionsForSubcommands.config.downloader)
 
         getProjectWithPackages(ortResult).forEach { (project, pkgList) ->
             val linkedReleases = pkgList.mapNotNull { pkg ->
                 val name = createReleaseName(pkg.id)
-                sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkg.id.version)
+                val release = sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkg.id.version)
                     .flatMap { sw360ReleaseClient.enrichSparseRelease(it) }
                     .orElseGet { createSw360Release(pkg, sw360ReleaseClient) }
+
+                if (attachSources) {
+                    val tempDirectory = createOrtTempDir(pkg.id.toPath())
+                    try {
+                        // First, download the sources of the package into a sources directory, whose root directory is
+                        // temporary.
+                        val sourcesDirectory = tempDirectory.resolve("sources")
```

Comment:
Note that theoretically it would not even be required to put sources and the ZIP at different directory levels, as the `archive()` function is smart enough to not ZIP the ZIP itself if sources and ZIP are in the same directory (see the `fileFilter = { it != zipFile }`). But I see how relying on this feature can be a bit confusing, so I'm also fine with keeping this as-is; your call.

Reply:
I prefer to split the sources from the zip file although the `archive()` function is smart enough.